buildtools/dep/dep.go

```diff
@@ -13,6 +15,25 @@ import (
 
 // ErrNoLockfile is returned if a dep manifest is found without an accompanying lockfile.
 var ErrNoLockfile = errors.New("found Gopkg.toml but not Gopkg.lock")
+var ErrIsIgnored = errors.New("this is ignored")
+
+// A Resolver is a wrapper for Lockfile to allow for dep specific toml file behavior. Inherits resolvers
+type Resolver struct {
+	TOMLFile TOMLFile
+	Lockfile Lockfile
+}
+
+// TOMLFile contains the contents of a dep toml file. TOMLFiles are resolvers.
+// Ignored is a result of toml unmarshalling. IgnoreMap is for easy search of ignored packages
+type TOMLFile struct {
```

Comment:
I would rather call this `Manifest`. `TOMLFile` is general and a bit misleading (this data structure does not represent _all_ TOML files).

In general, a "manifest" is a value describing the components in a project and a "lockfile" describes resolved revisions of components.